llm/llama.go

```diff
@@ -607,98 +610,116 @@ func (llm *llama) Predict(ctx context.Context, predict PredictOpts, fn func(Pred
 		request["grammar"] = jsonGrammar
 	}
 
-	// Handling JSON marshaling with special characters unescaped.
-	buffer := &bytes.Buffer{}
-	enc := json.NewEncoder(buffer)
-	enc.SetEscapeHTML(false)
+	for retries := 0; retries < maxRetries; retries++ {
```

Comment:
This diff is hard to read, I put the previous logic in a for loop. I'll point out the new logic.